ray-operator/controllers/ray/authentication_controller.go

```diff
@@ -103,13 +108,23 @@ func (r *AuthenticationController) Reconcile(ctx context.Context, req ctrl.Reque
 	rayCluster := &rayv1.RayCluster{}
 	if err := r.Get(ctx, req.NamespacedName, rayCluster); err != nil {
 		if errors.IsNotFound(err) {
-			logger.Info("RayCluster not found, likely deleted")
+			logger.Info("RayCluster not found, likely deleted - checking for orphaned resources")
+			// Clean up any orphaned ReferenceGrants in this namespace
+			if err := r.cleanupOrphanedReferenceGrant(ctx, req.Namespace, logger); err != nil {
+				logger.Info("Failed to cleanup orphaned ReferenceGrant (non-fatal)", "error", err)
+			}
 			return ctrl.Result{}, nil
 		}
 		logger.Error(err, "Failed to get RayCluster")
 		return ctrl.Result{RequeueAfter: MediumRequeueDelay}, err
 	}
 
+	// Handle deletion with finalizer
+	if !rayCluster.DeletionTimestamp.IsZero() {
```

Comment:
we're introducing a finalizer here. This is required because ownerreference can't be used cross namespace.